simtools/corsika/corsika_output.py

```diff
@@ -976,44 +1036,416 @@ def num_photons_per_event(self):
+                "title": "Photons per telescope distribution",
+                "edges": "Telescope counter",
+                "edges unit": u.dimensionless_unscaled,
+            },
+        }
+        return self.__dict_1D_distributions
+
+    def _export_1D_histograms(self, output_dir="./"):
+        """
+        Auxiliary function to export only the 1D histograms.
+
+        Parameters
+        ----------
+        output_dir: str
+            Output directory, where to save the histograms.
+        """
+
+        for _, function_dict in self._dict_1D_distributions.items():
+            self._meta_dict["Title"] = function_dict["title"]
+            function = getattr(self, function_dict["function"])
+            hist_1D_list, x_edges_list = function()
+            x_edges_list = x_edges_list * function_dict["edges unit"]
+            if function_dict["function"] == "get_photon_density_distr":
+                histogram_value_unit = 1 / (function_dict["edges unit"] ** 2)
+            else:
+                histogram_value_unit = u.dimensionless_unscaled
+            hist_1D_list = hist_1D_list * histogram_value_unit
+            for i_histogram, _ in enumerate(x_edges_list):
+                if self.individual_telescopes:
+                    ecsv_file = (
+                        f"{function_dict['file name']}_"
+                        f"tel_index_{self.telescope_indices[i_histogram]}.ecsv"
+                    )
+                else:
+                    ecsv_file = f"{function_dict['file name']}_all_tels.ecsv"
+
+                table = self.fill_ecsv_table(
+                    hist_1D_list[i_histogram],
+                    x_edges_list[i_histogram],
+                    None,
+                    function_dict["edges"],
+                    None,
+                )
+                ecsv_file = Path(output_dir).joinpath(ecsv_file)
+                self._logger.info(f"Exporting histogram to {ecsv_file}.")
+                table.write(ecsv_file, format="ascii.ecsv", overwrite=True)
+
+    @property
+    def _dict_2D_distributions(self):
+        """
+        Dictionary to label the 2D distributions according to the class methods.
+
+        Returns
+        -------
+        dict:
+            The dictionary with information about the 2D distributions.
+        """
+        if self.__dict_2D_distributions is None:
+            self.__dict_2D_distributions = {
+                "counts": {
+                    "function": "get_2D_photon_direction_distr",
+                    "file name": "hist_2D_photon_direction_distr",
+                    "title": "Incoming directive cosines",
+                    "x edges": "x directive cosinus",
+                    "x edges unit": u.dimensionless_unscaled,
+                    "y edges": "y directive cosinus",
+                    "y edges unit": u.dimensionless_unscaled,
+                },
+                "density": {
+                    "function": "get_2D_photon_time_altitude",
```

Comment:
This function is time/altitude but the name is "density".

Reply:
Corrected now